packages/editor/src/extensions/outline-list-item/outline-list-item.ts

```diff
@@ -140,17 +146,34 @@ export const OutlineListItem = Node.create<ListItemOptions>({
         if (!(e.target instanceof HTMLParagraphElement)) return;
         if (!li.classList.contains("nested")) return;
 
+        const { x, y, right } = li.getBoundingClientRect();
+
         const clientX =
           e instanceof MouseEvent ? e.clientX : e.touches[0].clientX;
+
         const clientY =
           e instanceof MouseEvent ? e.clientY : e.touches[0].clientY;
 
-        const { x, y } = li.getBoundingClientRect();
-
         const hitArea = { width: 26, height: 24 };
+
+        const selection = editor.state.selection;
+        const parent = findParentNode((node) => !!node.attrs.textDirection)(
+          selection
+        );
+
+        let xlowerLimit = clientX >= x - hitArea.width;
+        let xHigherLimit = clientX <= x;
```

Comment:
1. Can we rename these to `xStart` & `xEnd`?
2. It'd be better for readability to put the `y` coordinates here as well.